runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/SparkBeamMetric.java

```diff
@@ -38,9 +38,35 @@
 /**
  * An adapter between the {@link MetricsContainerStepMap} and Codahale's {@link Metric} interface.
  */
-class SparkBeamMetric implements Metric {
+public class SparkBeamMetric implements Metric {
   private static final String ILLEGAL_CHARACTERS = "[^A-Za-z0-9-]";
 
+  public static Map<String, String> renderAll(MetricResults metricResults) {
```

Comment:
What's the difference between this and the existing `renderAll` method?

Reply:
We need `Map<String, String> ` in Spark History Server, I created a new method renderAllToString and reused renderAll logic in it.

Reply:
Nice 👍